mypyc/genops.py

```diff
@@ -2662,6 +2801,13 @@ def call(self, decl: FuncDecl, args: Sequence[Value],
         return self.add(Call(decl, args, line))
 
     def visit_call_expr(self, expr: CallExpr) -> Value:
+        # Annoying special case for dataclasses 'field' function calls because
+        # the mypy dataclass plugin typechecks such a call using the types
+        # of the parameters to the default and default_factory
+        # arguments, resulting in attempted coercions that throw a runtime error.
+        if isinstance(expr.callee, NameExpr) and expr.callee.fullname == 'dataclasses.field':
+            self.types[expr] = AnyType(TypeOfAny.from_error)
```

Comment:
Could this be done with mypyc's `@specialize_function` mechanism instead of inline in `visit_call_expr`?